Lib/webbrowser.py

```diff
@@ -667,33 +676,35 @@ def open(self, url, new=0, autoraise=True):
             return True
 
 
-def main():
-    import getopt
-    usage = """Usage: %s [-n | -t | -h] url
-    -n: open new window
-    -t: open new tab
-    -h, --help: show help""" % sys.argv[0]
-    try:
-        opts, args = getopt.getopt(sys.argv[1:], 'ntdh',['help'])
-    except getopt.error as msg:
-        print(msg, file=sys.stderr)
-        print(usage, file=sys.stderr)
-        sys.exit(1)
+def parse_args(arg_list: list[str] | None):
+    import argparse
+    parser = argparse.ArgumentParser(description="Open URL in a web browser.")
+    parser.add_argument("url", help="URL to open")
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-n", "--new-window", action="store_true",
+                       help="open new window")
+    group.add_argument("-t", "--new-tab", action="store_true",
+                       help="open new tab")
+
+    args = parser.parse_args(arg_list)
+
+    return args
+
+
+def main(arg_list: list[str] | None = None):
+    args = parse_args(arg_list)
+
     new_win = 0
-    for o, a in opts:
-        if o == '-n': new_win = 1
-        elif o == '-t': new_win = 2
-        elif o == '-h' or o == '--help':
-            print(usage, file=sys.stderr)
-            sys.exit()
-    if len(args) != 1:
-        print(usage, file=sys.stderr)
-        sys.exit(1)
-
-    url = args[0]
-    open(url, new_win)
+    if args.new_window:
+        new_win = 1
+    elif args.new_tab:
+        new_win = 2
```

Comment:
I believe this could be simplified by having the two options use `action='store_const'` to a common `dest`.

Not that this is "complex", but it does seem more verbose than it need be :)

Reply:
Yes, much cleaner :)

Reply:
I was wondering if the constants could be stored directly.  The condensed change looks even better to me also.  I am thinking of doing getopt to argparse for IDLE.